### DISCUSSION TITLE
Inserting new page into document rendered to both html/pdf and revealjs

### DISCUSSION THREAD

**Discussion by tomkom** [quarto-dev/quarto-cli#4721] [Q&A]
have a project with content that adaptively renders subset of content to presentations (with exercises for students) and a subject "book".

Sometimes the long format of slides is a bit of a pain, and I would like to include a page break. `---` does not seem to alwyas work, or renders a line in the output text, and causes issues for the html/pdf pipeline. Inserting a blank slide wtih a blank header works `##`, but then includes an unsightly numbered section with no content in the html/pdf output.

I have found somewhere {{new page}} ( or some such similar command, cannot recall) that just did not work. Any hint?
Presumably I could surround 

```
:::{content visible when-format="revealjs"}

##

:::
```
but that seems like overkill

Martin

**Comment by cderv** [comment]
> Sometimes the long format of slides is a bit of a pain, and I would like to include a page break.

I don't think HTML pagebreak means something in a HTML presentaton. That is why the pagebreak feature does not work for revealjs format. 

We could adapt it to create a new slide in presentation, but it is not the same meaning, and not the same implementation. 

I think what you want is to add a new slide at some point using some syntax that would apply only in revealjs. 

The syntax you found is the way to do it currently, but you could also use a shorter syntax and create a lua filter for your project. Example 

````markdown
---
title: test
format: 
  html: default
  revealjs: default
filters: [new-slide.lua]
keep-md: true
---

## content 

Some content 

## {.only-reveal}

Some other content on a new slide

````

Lua filter
```lua
Header = function(h)
  if h.classes:includes("only-reveal") then
    if not quarto.doc.is_format("revealjs") then
      return {}
    else
      h.classes["only-reveal"] = nil
    end
  end
end
```

This filter could be apply to your whole project, and it removes the header except on revealjs, where it will become a new slide

We have some docs to help developing in Lua https://quarto.org/docs/extensions/lua.html 

Hope it helps

**Reply by tomkom** [reply]
Uh, this is a bit advanced... I am not sure I can sink my teeth into this. I have no idea (even based on reading the docs) where I would put this lua code... I know it is used, and I am fine to learn the syntax, but the installation to lua filters is not documented intuitively enough for me. I will try ith the longer code I showed. It seems ignored at times, though.

**Reply by cderv** [reply]
Oh sorry, I did not explain as I thought the YAML indication would be enough 
* I added this in your YAML header
```yaml
filters: [new-slide.lua]
```
* the `new-slide.lua` leaves right next to your document or in your project folder if you defined your fomat in `_quarto.yaml`

This is the less advanced way of using Lua filters. The next step would be to create an Extension so that this type of feature can be reused, and could be improved if necessary with parameters, etc... But you don't need that here. 

Hopefully creating a lua file and setting the config in the YAML is as straightforward as I think it is. Lua filter is the way to program your document content structure, but your way of using conditional `when-format` is a good way too for what you need. Using snippet feature in an IDE you work with would help you insert quickly this piece of code. 

Anyhow, hope it helps you
